middleware/webhooks.js

```diff
@@ -85,15 +86,13 @@ async function parseUserAction(entry) {
 
     // Postbacks and quick replies are handled in exactly the same way, the payload
     // is just in a different location in the object.
-    const payload = event.postback
-      ? event.postback.payload
-      : event.message.quick_reply.payload;
+    const payload = event.postback ? event.postback.payload : event.message.quick_reply.payload;
 
     parsed_data =
       event.postback && event.postback.referral
         ? {
             ...parsed_data,
-            ...queryStringToObject(event.postback.referral.ref),
+            ...queryStringToObject(),
```

Comment:
queryStringToObject takes an argument so this looks it'll return an empty object then spread nothing in.  So I think we need `event.postback.referral.ref` here